adr/ADR-32.md

```diff
@@ -216,6 +229,8 @@ only returned if the `schema` was specified when created.
 }
 ```
 
+The type for this is `io.nats.micro.v1.stats_response`.
```

Comment:
Since the stats can be modified - the stats handler can attach additional data, should this be customizable by the stats handler?

Reply:
Maybe though the `data` is free form so if the handler wants to mark it up its welcome to mark it up, in the schema I'll just say its anything?

Reply:
@ripienaar What do you mean by "free form" I have already suggested that data just be a string, That string could be whatever, it could happen to be json, it could happen to be the word "true", it could happen to be a number. Just like our message payload, just like our kv value. 
As it is now, we had to build a complicated way to inject a data handler to handle a json anything, whereas if it is just string, all I have to worry about is make sure I json encode the string properly since it's a string in a json object.

Reply:
```
returns unknown data that can be serialized as JSON
```

it's anything.  So I wont mark up the specific type of the included data, the type here only conveys that its a micro status message.  What random stuff people put in the extra data is their business

Reply:
The doc is not clea or complete, `data` is not "anything", it is json, meaning it is expected that the right side of the `"data":` is an actual json object, array, number or boolean, string etc., but not "just" a string.
This is why it requires a handler to encode and decode and this is why it's a huge pain. Not all languages have built in support for json, which is what this requires because it means passing a generic language specific json object to the user (or in the case of java/.net I have to send a string of the contents to the encoder/decoder which is double processing) otherwise requiring a callback to handle. 
It would be easier / less complex to code and to understand if it `data`'s type was string, instead of being fancy.

Reply:
The spec is clear, something that can be serialized as JSON, it implies its valid JSON.

I have no idea why you are discussing the data field on this PR its unrelated. I am adding a type hint for the Stats object.

Reply:
I bring it up because the idea of a generic `type` is parallel to `data`, and if a generic string is good enough for `type` why isn't it good enough for `data`

Reply:
> Maybe though the `data` is free form so if the handler wants to mark it up its welcome to mark it up, in the schema I'll just say its anything?

Reply:
The type describes this:

```go
	Stats struct {
		ServiceIdentity
		NumRequests           int             `json:"num_requests"`
		NumErrors             int             `json:"num_errors"`
		LastError             string          `json:"last_error"`
		ProcessingTime        time.Duration   `json:"processing_time"`
		AverageProcessingTime time.Duration   `json:"average_processing_time"`
		Started               string          `json:"started"`
		Data                  json.RawMessage `json:"data,omitempty"`
	}
```

There's a LOT to describe and derive etc and much to use in addition to the `data` and the kind describes this.

Reply:
That's from the go client impl. It's nice and convenient for go to have that built in type. Java and .NET do not have an equivalent or even a json interface to lean on, so those clients have quite a bit of complicated work to simulate that.

Reply:
`json.RawMessage` isnt anything special, its just `[]byte` that holds valid JSON.  If Java can validate something is valid JSON you can a) get from stats handler a string b) validate its JSON c) store it as `string` in the data.

Reply:
Java and .NET cannot natively validate something is json.